### ISSUE TITLE
Don't trigger "MayBeConst" rule on "actual val" in KMP projects.

### ISSUE THREAD

**Issue by gtcompscientist** [detekt/detekt#4330]
## Expected Behavior of the rule
`MayBeConst` rule unexpectedly flags Kotlin Multiplatform files using the `expect`/`actual` pattern at the `actual` site.

Example failure:
File 1:
```
expect val stringVal: String
```
File 2:
```
actual val stringVal: String = "abc123" // detekt says "MayBeConst", but it cannot.
```

## Context

Want to be able to use this rule on KMP projects without turning this rule off, or requiring people suppress it everywhere, as there are still lots of valid places to trigger this rule.

Reference for expect/actual: https://kotlinlang.org/docs/mpp-connect-to-apis.html
